tests/tools/TestApp/src/input.c

```diff
@@ -48,3 +57,210 @@ char *parse_json_to_string(const char *file_name) {
     // Return the buffer as a const char*
     return buffer;
 }
+
+int get_video_params(const char *json_string, video_params *params) {
+
+    json_t *root;
+    json_error_t error;
+    json_t *video_value;
+    int err = 0;
+
+    root = json_loads(json_string, 0, &error);
+    if (!root) {
+        fprintf(stderr, "error: on line %d: %s\n", error.line, error.text);
+        json_decref(root);
+        err = 1;
+        goto exit;
+    }
+
+    json_t *payload = json_object_get(root, "payload");
+    if (!payload) {
+        fprintf(stderr, "error: key not found\n");
+        json_decref(payload);
+        err = 1;
+        goto exit;
+    }
+    json_t *video = json_object_get(payload, "video");
+    if (!video) {
+        fprintf(stderr, "error: key not found\n");
+        json_decref(video);
+        err = 1;
+        goto exit;
+    }
+    video_value = json_object_get(video, "fps");
+    if (!video_value) {
+        fprintf(stderr, "error: key not found\n");
+        json_decref(video_value);
+        err = 1;
+        goto exit;
+    }
+    params->fps = json_number_value(video_value);
+
+    video_value = json_object_get(video, "pixelFormat");
+    if (!video_value) {
+        fprintf(stderr, "error: key not found\n");
+        json_decref(video_value);
+        err = 1;
+        goto exit;
+    }
+    char *temp_pix_format = (char *)json_string_value(video_value);
+    if (strcmp(temp_pix_format, "yuv422p10le") == 0) {
+        params->pixel_format = MESH_VIDEO_PIXEL_FORMAT_YUV422PLANAR10LE;
+    } else if (strcmp(temp_pix_format, "v210") == 0) {
+        params->pixel_format = MESH_VIDEO_PIXEL_FORMAT_V210;
+    } else if (strcmp(temp_pix_format, "yuv422p10rfc4175") == 0) {
+        params->pixel_format = MESH_VIDEO_PIXEL_FORMAT_YUV422RFC4175BE10;
+    } else {
+        fprintf(stderr, "error: invalid pixel format\n");
+        goto exit;
+    }
+
+    video_value = json_object_get(video, "width");
+    if (!video_value) {
+        fprintf(stderr, "error: key not found\n");
+        json_decref(video_value);
+        err = 1;
+        goto exit;
```

Comment:
Please replace block of code 
`err = 1;
 goto exit;
`
With `return 1`
You can also remove `goto exit`

Reply:
fixed